core/src/storage/impls/storage_db/snapshot_db_manager_sqlite.rs

```diff
@@ -98,13 +98,16 @@ impl SnapshotDbManagerSqlite {
         old_snapshot_epoch_id: &EpochId, delta_mpt: &DeltaMptInserter,
     ) -> Result<MerkleHash>
     {
-        let maybe_old_snapshot_db = SnapshotDbSqlite::open(
-            &self.get_snapshot_db_path(old_snapshot_epoch_id),
-        )?;
-        let mut old_snapshot_db = maybe_old_snapshot_db
-            .ok_or(Error::from(ErrorKind::SnapshotNotFound))?;
-
-        temp_snapshot_db.copy_and_merge(&mut old_snapshot_db, delta_mpt)
+        if *old_snapshot_epoch_id == NULL_EPOCH {
```

Comment:
Instead of doing it here, it's better to add a if-branch before `if self.try_make_cow_copy(`